### PR TITLE
IDOC Satellite #minor

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Add option for IODC satellite pull
Main changes are in app.py, other changes are small refactoring and small changes needed for IODC

Refactor `use_backup` to `use_hr_serviri` in many places

#279 

Example of image - 

Screenshot 2024-08-29 at 11 29 54

and windy.com weather roughly at the same time 
Screenshot 2024-08-28 at 22 57 17

Q: do we see this white band in the satellite image that we trained on?

Im trying to work out the right balance of re-writing lots of things, and get this working. Have discussed with @devsjc about re-writing this consumer completely

## How Has This Been Tested?

CI + extra tests

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: satip/app.py [hunk 1]
```diff
@@ -104,8 +121,42 @@ def run(
...
     start_time: str = pd.Timestamp.utcnow().isoformat(timespec="minutes").split("+")[0],
     cleanup: bool = False,
-    use_backup: bool = False,
+    use_hr_serviri: bool = False,
     maximum_n_datasets: int = -1,
+    use_iodc: bool = False,
+):
+    """ See below for function description.
+
+    There is slight duplicate, but testing adn debugging is easier with this setup.
```
Comment (peterdudfield) [inline satip/app.py:130]: typo

File: satip/utils.py [hunk 2]
```diff
@@ -838,14 +855,18 @@ def move_older_files_to_different_location(save_dir: str, history_time: pd.Times
             continue
         if "hrv" in date:
             file_time = pd.to_datetime(
-                date.replace("15_", "").split(".zarr.zip")[0].split("/")[-1].split("_")[-1],
+                date.replace("iodc_", "")
+                .replace("15_", "")
+                .split(".zarr.zip")[0]
+                .split("/")[-1]
+                .split("_")[-1],
```
Comment (peterdudfield) [inline satip/utils.py:862]: im tempted to refactor this

**Comment by Sukh-P** [comment]
So for reference this is what the IODC image looked like that we pulled from HF and what we cropped down to to get just India to train on 
Screenshot 2024-07-18 at 17 02 53
Screenshot 2024-07-19 at 12 20 59

**Comment by peterdudfield** [comment]
> So for reference this is what the IODC image looked like that we pulled from HF (sorry not got a better screenshot of the one that shows India a bit more clearly on it) and what we cropped down to to get just India to train on Screenshot 2024-07-18 at 17 02 53 Screenshot 2024-07-19 at 12 20 59

Thanks, that looks different enough, I think somethin may not be right with the PR, let me look in to it

**Comment by peterdudfield** [comment]
> > So for reference this is what the IODC image looked like that we pulled from HF (sorry not got a better screenshot of the one that shows India a bit more clearly on it) and what we cropped down to to get just India to train on Screenshot 2024-07-18 at 17 02 53 Screenshot 2024-07-19 at 12 20 59
> 
> Thanks, that looks different enough, I think somethin may not be right with the PR, let me look in to it

This is now sorted, thank you @Sukh-P for this input

File: satip/app.py [hunk 3]
```diff
@@ -81,8 +91,8 @@
     type=click.BOOL,
 )
 @click.option(
-    "--use-backup",
-    envvar="USE_BACKUP",
+    "--use-hr-serviri",
+    envvar="USE_HR_SERVIRI",
```
Comment (devsjc) [inline satip/app.py:95]: So is this the same service as the backup, but we're just being more explicit?

File: satip/app.py [hunk 4]
```diff
@@ -141,37 +192,60 @@ def run(
...
-                start_date=start_date.strftime("%Y-%m-%d-%H:%M:%S"),
-                end_date=pd.Timestamp(start_time, tz="UTC").strftime("%Y-%m-%d-%H:%M:%S"),
-            )
-            # Check if any RSS imagery is available, if not, fall back to 15 minutely data
-            if (len(datasets) == 0) or use_backup:
-                log.warn(
-                    f"No RSS Imagery available or using backup ({use_backup=}), "
-                    f"falling back to 15-minutely data",
+
+            if not use_iodc:
```
Comment (devsjc) [inline satip/app.py:201]: Usually easier to follow the positive logic flow

File: tests/integration_test/test_integration_app.py [hunk 5]
```diff
@@ -20,50 +20,56 @@ def test_save_to_netcdf():  # noqa 103
...
+
+@freeze_time("2024-08-28 12:00:00")  # Date with IODC
+def test_iodc():  # noqa 103
+
+    from satip import utils
+    utils.setupLogging()
+
+    user_key = os.environ.get("EUMETSAT_USER_KEY")
+    user_secret = os.environ.get("EUMETSAT_USER_SECRET")
+    with tempfile.TemporaryDirectory() as tmpdirname:
```
Comment (devsjc) [inline tests/integration_test/test_integration_app.py:44]: Does "/tmp" get mapped out as a volume in the container? Just conscious that a lot of data could get dumped into it

**Review by devsjc — APPROVED** [review]

File: satip/app.py (hunk 3 above)
Comment (peterdudfield) [reply satip/app.py:95]: yea exactly, I was getting confused, so I wanted to write it properly

File: tests/integration_test/test_integration_app.py (hunk 5 above)
Comment (peterdudfield) [reply tests/integration_test/test_integration_app.py:44]: yea, im no sure where it goes, but it gets deleted after the test

File: satip/app.py (hunk 4 above)
Comment (peterdudfield) [reply satip/app.py:201]: yea good point, ill try to change this

**Pull request merged**